### DISCUSSION TITLE
Please document RMMmuMemoryMap :)

### DISCUSSION THREAD

**Discussion by Geemahn** [NVIDIA/open-gpu-kernel-modules#633] [Q&A]
Hi there, I would appreciate it if I could get documentation on the NVreg setting RMMmuMemoryMap. I am dealing with what I think is a resource allocation issue and am extremely curious if this setting is capable of addressing it ( pun intended ;P).

**Comment by mtijanic** [comment]
It doesn't do anything with the open kernel modules, so nothing to document really. You might want to poke at some of the object properties directly to debug your issue, such as maybe setting this to true:
https://github.com/NVIDIA/open-gpu-kernel-modules/blob/550.78/src/nvidia/generated/g_virt_mem_allocator_nvoc.h#L124

Good luck!

**Reply by Geemahn** [reply]
I appreciate your thoughtful reply Milos. I am not a programmer but taking your suggestion into consideration I edited this line https://github.com/NVIDIA/open-gpu-kernel-modules/blob/083cd9cf17ab95cd6f9fb50a5349c21eaa2f7d4b/src/nvidia/generated/g_virt_mem_allocator_nvoc.c#L153 to change the '==' to '!=' then compiled the modules and installed them. My die is listed in the ChipHal comment two lines up.

Was that effective or proper? 
Do you have a suggested way to accomplish enabling this or any PDB_PROP? 
Is there a way I can force such a variable during boot similar to NVreg Dwords without having to edit and recompile the modules?
Is there a way to verify such a setting in a booted environment?

Thank you for your guidance :)

**Reply by mtijanic** [reply]
Oh, my bad, I didn't realize it was already TRUE\*. Yeah, setting that to false is not going to have any meaningful effect. FALSE is only used for some internal simulation tools where the whole codebase is actually compiled in userspace.

Whatever your issue is, this isn't gonna help. Maybe best just open an issue and we'll triage and look at it (although it might take a while, sorry).


\* sidenote: we really need to improve the generated output to just say `NV_TRUE` instead of `((NvBool)(0 == 0))`, that's hardly readable

**Reply by Geemahn** [reply]
Before I could open an issue I would need concrete metrics to guide discovery. Is there a way I can use RmLockTimeCollect to record my lock times since I have noticed that RMPriorityBoost improves responsiveness on this system?

Since metering lock times has been out of my reach I instead used Nsight Systems to look at possibly excessive GPU context switches however I was never able to get any advice on the topic from the developer forum https://forums.developer.nvidia.com/t/context-switching-overload/287663?u=jwick

Your replies do give me the courage to continue poking round in the modules and I will explore more of the PDB properties.

Thanks for your continued help.
